### ISSUE TITLE
create mechanism to load demo content

### ISSUE THREAD

**Issue by htpvu** [jhu-idc/iDC-general#307]
Migrations in the `demo` directory (to be done in #306) are intended as seed content for developer convenience when poking at the development environment.  If there is a simple and straightforward way to conveniently load these migrations as a simple `make` target, we ought to have that available for added convenience.  If this ends up being a challenge, it may not be worth the effort to pursue this task.

**Comment by emetsger** [comment]
I seem to be being assigned things :)

A couple questions: is "demo" content the same content that we discussed as @bseeger's "convenience data" at the most recent IDC developer meeting?  If so, can I be reminded of the rationale to write code to import data when a migration module (i.e. writing no code) would work just as well (not to mention that is exactly the purpose a migration module is fit for)?  I probably wasn't paying to close attention to that part of the meeting, I'll admit, because I didn't expect to get assigned the work, so I didn't really have skin in the game so to speak.

**Comment by birkland** [comment]
Yes, this the "convenience data".  The idea of a loading harnass (i.e. a `make` target that will slurp in all such content) is merely to make it convenient and easy to get this data in.  i.e.  something like `make populate-demo-data` vs "login, navigate, click, select migration, find csv in file selector, click; repeat for all desired files"  This would be a low-priority low-effort recycling of the pattern used in migration test for loading the CSVs, and would happen some time after #306 is done.

**Comment by emetsger** [comment]
Ok, but a make target could just as easily perform a migration using a migrate module, no?  Retains the benefit of doing it on an as-needed basis without writing any code.

**Comment by birkland** [comment]
Whichever approach is simplest, easiest, and most straightforward is the right one.  I edited the title to delete `testcafe`, and fleshed out the description.  Using testcafe to upload CSVs seemed to be the easiest on the call, but if a migrate module is easier, great!

**Comment by emetsger** [comment]
Sounds good.  Yeah, a migration module is _exactly_ fit for this use case, and we wouldn't  have to write any additional code to maintain it.

**Comment by bseeger** [comment]
If we auto load all data from the upcoming `demo` directory, then the assumption is that all the data in `demo` dir is cohesive, meaning that it all works together to build a data set that's somewhat coherent.   I'm okay with that, but would also like to have some files that provide different scenarios - like a large number of collections or items.   

Should this data be one cohesive set?  Or can we have a few sub folders - one of which is a cohesive set that can be auto loaded via a make target?

**Comment by emetsger** [comment]
At this juncture, the plan is to place demo data in a migration module.  Tags could be used to distinguish between different data loads, and those tags could be exposed to the developer as make arguments.
